app/src/main/java/com/conupods/Settings/SettingsActivity.java

```diff
@@ -0,0 +1,107 @@
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.CheckBox;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.conupods.OutdoorMaps.View.MapsActivity;
+import com.conupods.R;
+
+public class SettingsActivity extends AppCompatActivity {
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.settings_page);
+
+        Button done = findViewById(R.id.done2);
+        Button personalPage = findViewById(R.id.toggle2_1);
+        Button infoPage = findViewById(R.id.toggle2_2);
+
+        CheckBox metro = findViewById(R.id.metro);
+        CheckBox train = findViewById(R.id.train);
+        CheckBox bus = findViewById(R.id.bus);
+        CheckBox concordiaShuttle = findViewById(R.id.concordiaShuttle);
+
+        CheckBox elevators = findViewById(R.id.elevators);
+        CheckBox escalators = findViewById(R.id.escalators);
+        CheckBox stairs = findViewById(R.id.stairs);
+        CheckBox accessibilityInfo = findViewById(R.id.accessibilityInfo);
+        CheckBox stepFreeTrips = findViewById(R.id.stepFreeTrips);
+
+        setCheckedBoxes();
+
+        //Top Menu Button
+
+        done.setOnClickListener(view -> {
+            startActivityIfNeeded(new Intent(SettingsActivity.this, MapsActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0);
+        });
+        personalPage.setOnClickListener(view -> {
+            startActivityIfNeeded(new Intent(SettingsActivity.this, SettingsPersonalActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0);
+        });
+        infoPage.setOnClickListener(view -> {
+            startActivityIfNeeded(new Intent(SettingsActivity.this, SettingsInfoActivity.class).setFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT), 0);
+        });
+
+        //My transit options
+
+        metro.setOnClickListener(view -> changePreferences(metro));
+        train.setOnClickListener(view -> changePreferences(train));
+        bus.setOnClickListener(view -> changePreferences(bus));
+        concordiaShuttle.setOnClickListener(view -> changePreferences(concordiaShuttle));
+
+        //My indoor options
+
+        elevators.setOnClickListener(view -> changePreferences(elevators));
+        escalators.setOnClickListener(view -> changePreferences(escalators));
+        stairs.setOnClickListener(view -> changePreferences(stairs));
+        accessibilityInfo.setOnClickListener(view -> changePreferences(accessibilityInfo));
+        stepFreeTrips.setOnClickListener(view -> changePreferences(stepFreeTrips));
+    }
+
+    protected void changePreferences(CheckBox preference) {
+
+        SharedPreferences preferences = getApplicationContext().getSharedPreferences("Preferences", Context.MODE_PRIVATE);
+        Editor prefEdit = preferences.edit();
+
+        if (preference.isChecked()) {
+            prefEdit.putBoolean(String.valueOf(preference.getId()), true).apply();
+        } else {
```

Comment:
new line for readability

Reply:
Not sure which line you mean exactly but I assume add a line before the method right? Correct me if I'm wrong

Reply:
I think Finn is talking about the else

Reply:
yes the else should get a new line